OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -57,6 +57,11 @@ public override Activity Tick(Actor self)
 			foreach (var attack in attackTraits.Where(x => !x.IsTraitDisabled))
 			{
 				var activity = InnerTick(self, attack);
+				if (activity != this && activity != NextActivity) {
+					// Attack switched from frozen target to real or vice versa
```

Comment:
We are in the process of splitting up activities into more modular parts, so I expect this assumption would cause us problems in the future.

I think in this case we might be better to remove `readonly` from `Target` and change that instead.